Add Normalize function

ahmedhus22 opened the pull request:
Closes #693

cliffburdick commented:
/build

cliffburdick commented:
> should i add the license at the start of new source files or its not necessary?

yes, please copy and paste from another file

cliffburdick commented on include/matx/transforms/normalize.h:
```diff
@@ -0,0 +1,113 @@
+////////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (c) 2021, NVIDIA Corporation
```
2025

cliffburdick reviewed (changes requested):
This looks great! Thanks for adding this. I left a couple minor comments.

ahmedhus22 commented on docs_input/api/linalg/other/normalize.rst:
```diff
@@ -0,0 +1,32 @@
+normalize
+===========
+
+normalize all values in a tensor according to specified method, equivalent to MATLAB's implementation
```
Added a table for available `NORMALIZE_RANGE` options and also a link to Matlab doc for normalize as its based on that

ahmedhus22 commented:
Addressed the requested changes, and fixed an issue with `NormalizeScale` test where I was calculating stdd across the whole tensor, even though it is only normalized across 0th dimension.

cliffburdick commented:
/build
